lockup-factory/src/lib.rs

```diff
@@ -0,0 +1,423 @@
+mod types;
+mod utils;
+
+use crate::utils::*;
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::{env, ext_contract, near_bindgen, AccountId, Promise, PromiseOrValue, Balance};
+use near_sdk::json_types::{U128};
+use near_sdk::serde::{Serialize};
+pub use crate::types::*;
+
+/// There is no deposit balance attached.
+const NO_DEPOSIT: Balance = 0;
+const TRANSFERS_STARTED: u64 = 1603274400000000000;
+
+#[global_allocator]
+static ALLOC: near_sdk::wee_alloc::WeeAlloc<'_> = near_sdk::wee_alloc::WeeAlloc::INIT;
+
+const CODE: &[u8] = include_bytes!("../../lockup/res/lockup_contract.wasm");
+
+
+pub mod gas {
+    use near_sdk::Gas;
+
+    /// The base amount of gas for a regular execution.
+    const BASE: Gas = 25_000_000_000_000;
+
+    /// The amount of Gas the contract will attach to the promise to create the lockup.
+    pub const LOCKUP_NEW: Gas = BASE;
+
+    /// The amount of Gas the contract will attach to the callback to itself.
+    /// The base for the execution and the base for cash rollback.
+    pub const CALLBACK: Gas = BASE;
+}
+
+const MIN_ATTACHED_BALANCE: Balance = 35_000_000_000_000_000_000_000_000;
+
+/// External interface for the callbacks to self.
+#[ext_contract(ext_self)]
+pub trait ExtSelf {
+    fn on_lockup_create(
+        &mut self,
+        lockup_account_id: AccountId,
+        attached_deposit: U128,
+        predecessor_account_id: AccountId,
+    ) -> Promise;
```

Comment:
Even though the type here is not important
```suggestion
    ) -> bool;
```